src/extension/features/toolkit-reports/pages/income-vs-expense/component.jsx

```diff
@@ -47,6 +48,16 @@ export class IncomeVsExpenseComponent extends React.Component {
 
   _masterCategoriesCollection = Collections.masterCategoriesCollection;
 
+  _localStorageKey = 'income-vs-expense-collapse-state';
+
+  constructor() {
+    super();
+
+    this.state = {
```

Comment:
bit confusing to have this in state, I don't think there's any reason to put this in state? maybe just store is as a variable on the class itself?

you can do this above too (outside of the constructor) with the other class props

Reply:
The collapsedState of a row was already a part of state prior to this modification. It's been a part of state since initial commit of this file. This state gets passed into child components for them to perform display logic based on the current state. Every time a row is collapsed, this component is going to need to be re-rendered and updated accordingly. I think it makes sense to keep it as a part of state.

Reply:
ahhh, I'm so used to putting `render` functions at the bottom these days that I didn't realize this copmonent was so much bigger, my bad. this should be fine then